### DISCUSSION TITLE
How do I set this up?

### DISCUSSION THREAD

**Discussion by CaliMiller** [chrisbenincasa/tunarr#712] [General]
Hello.

Today I'm setting up a homelab running Proxmox. This is my first time using Proxmox, Docker, and Tunarr. I'm a n00b.

I initially tried creating a container in Proxmox to run the binary, but I couldn't figure out how to get the binary into the container.

I then tried creating a new Linux VM and installed Docker. I was able to pull Tunarr in Docker. After awhile of tinkering, I was able to get it to run Tunarr, but the IP address for the http server is invalid, and it just freezes like this screenshot.

![Screenshot 2024-08-24 at 6 34 28 AM](https://github.com/user-attachments/assets/6ec895c4-b022-46a4-8420-8a0059635c3d)

I'm unable to proceed, and don't know what to do next.

**Comment by chrisbenincasa** [comment]
So this is a pretty large / general question that goes beyond things specific to Tunarr. I'll try to answer the best I can.

You can get the binary into your LXC simply by downloaded it via wget or curl from within the LXC. From there, the (undocumented) path would be to set up Tunarr as a system service. At some point I'd love to some actual packages that one could use to install Tunarr via the system's package manager (tracked here https://github.com/chrisbenincasa/tunarr/issues/267)

The other path you were working on (Docker in VM/LXC) seems to be working fine. Within the container, Tunarr binds itself to 0.0.0.0, which means it is listening on all network interfaces. In the context of Docker, this is necessary in order for requests to make it to Tunarr from within the container. You'll have to run a command similar to the example in https://tunarr.com/getting-started/setup/ in order to actually bind a port from your host (Docker LXC) to the Tunarr Docker container. At this point, you can access Tunarr by accessing the IP of your LXC with the port you chose.

So, for instance, if you ran the example in the docs, you'd go to `http://DOCKER_LXC_IP:8000` to access Tunarr.
